src/PrivateDns/PrivateDns/Records/NewAzurePrivateDnsRecordSet.cs

```diff
@@ -0,0 +1,145 @@
+{
+    using System.Collections;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.PrivateDns.Models;
+    using Microsoft.Azure.Management.PrivateDns.Models;
+    using Microsoft.Azure.Commands.PrivateDns.Utilities;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using ProjectResources = Microsoft.Azure.Commands.PrivateDns.Properties.Resources;
+
+    /// <summary>
+    /// Creates a new record set.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsRecordSet", SupportsShouldProcess = true, DefaultParameterSetName = FieldsParameterSetName),OutputType(typeof(PSPrivateDnsRecordSet))]
+    public class NewAzurePrivateDnsRecordSet : PrivateDnsBaseCmdlet
+    {
+        private const string FieldsParameterSetName = "Fields";
+        private const string ObjectParameterSetName = "Object";
+        private const string ResourceParameterSetName = "ResourceId";
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group to which the zone belongs.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The zone in which to create the record set (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/privateDnsZones", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string ZoneName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The PrivateDnsZone object representing the zone in which to create the record set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PSPrivateDnsZone Zone { get; set; }
+
+        [Parameter(ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the records in this record set (relative to the name of the zone and without a terminating dot).")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The type of Private DNS records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public RecordType RecordType { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The TTL value of all the records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public uint Ttl { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "A hash table which represents resource tags.")]
+        public Hashtable Metadata { get; set; }
```

Comment:
Why wouldn't you want this to be called Tag?

Reply:
Keeping it uniform between public and private dns